### PR TITLE
Feat - Move IOTA Protocol specific constants to constants.h and define macro for NODE_API_PATH "/api/v2"

### LINKED ISSUE #397: Feat - gathering protocol constats in headers
## Description

- [x] The protocol constants, like `IOTA_TRANSACTION_ID_BYTES`, `IOTA_MESSAGE_ID_BYTES`, are located in separated headers. the problem is users might not aware of those constants if they don't look into the implementation.
gathering constants in corresponding header files can have a clear view and prevent redefining constants, like  
* src/crypto/constants.h
* src/core/constants.h
* src/client/constants.h

the higher module should include sub-constants in its header file, so users can use the crypto constants by including client/constants.h
ex: crypto/constants.h should be included in core/constants.h and so on.

- [x] Move `/api/v2/` hardcoded api path from all endpoints to a common header in client folder.

## Motivation

Improve readability and usage of the library.

### PR DESCRIPTION
**Pull request by rebinmannady93**
# Description of change

Fixes #397 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

test/client/api_restful/test_get_xxx.c

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by matjazv — APPROVED** [review]
Nicely done!

The only question I have is which constants/defines should also be moved from different header files to a newly created `constants.h` files.
For example:
- in `address.h` there are some defines like `#define ADDRESS_MAX_BYTES ED25519_PUBKEY_BYTES`
- in `utxo_input.h` there is a define `#define UTXO_INPUT_MAX_COUNT 128`
- etc...

So should those defines stay where the currently are, or should they be moved to `constants.h` file.

File: src/crypto/constants.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef __CRYPTO_CONSTANTS_H__
+#define __CRYPTO_CONSTANTS_H__
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
```
Comment (oopsmonk) [inline src/crypto/constants.h]: remove, unused headers

File: src/client/constants.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef __CLIENT_CONSTANTS_H__
+#define __CLIENT_CONSTANTS_H__
+
+#include "core/constants.h"
+#include "crypto/constants.h"
```
Comment (oopsmonk) [inline src/client/constants.h]: remove, `crypto/constants.h` has included in `core/constants.h`
```suggestion

```

File: src/crypto/constants.h (hunk 1 above)
Comment (rebinmannady93) [reply src/crypto/constants.h]: Done.

File: src/client/constants.h (hunk 2 above)
Comment (rebinmannady93) [reply src/client/constants.h]: Done.

**Review by oopsmonk — APPROVED** [review]

**Review by matjazv — APPROVED** [review]

**Pull request merged**